src/corecel/math/detail/Sincospi.hh

```diff
@@ -0,0 +1,206 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2023-2024 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file corecel/math/detail/Sincospi.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include <cmath>
+#include <cstdint>
+
```

Comment:
```suggestion
#include "corecel/Config.hh"

#ifdef CELERITAS_SINCOSPI_PREFIX
#error "sincospi is provided natively on this platform. Do not include Sincospi.hh"
#endif
```

Reply:
What if we want to test the implemented functions? You should only `#error` to prevent further build errors...

Reply:
The problem is that if a user 'inadvertently' use this, there will be duplicate symbols or worse simple silent break of ODR (eg. if all the sincospi get inlined).  If one wants to test those function on a platform that already has `sincospi` additional care/gymnastic is anyway needed and it might make sense for that test to have to explicitly `#undef` the macro.

Reply:
No there won't; the `detail` functions in this header are uniquely defined. The only potential ODR violation will be if `::celeritas::sincospi` (for example) calls one implementation in one translation unit and another implementation in another. No macro will fix that, and since they're forceinlined it probably won't double-define anyway.

Reply:
> since they're forceinlined it probably won't double-define anyway.

That the worst case scenario (it silent weird result because the 2 implementation give slightly different result).

> No macro will fix that

Sort of.  I indeed completely missed that this header is defining `sincospi_impl` rather than`sincospi`.  So indeed it would take a corrupted build using 2 distinct config header file to trigger the problem.

So we can leave it as is.